modules/components/src/AdvancedFacetView/aggComponents.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { TermAgg, RangeAgg, BooleanAgg, DatesAgg } from '../Aggs';
+import { currentFieldValue } from '../SQONView/utils';
+import { inCurrentSQON } from '../SQONView/utils';
+
+const composedTermAgg = ({ sqon, onValueChange, ...rest }) => (
+  <TermAgg
+    maxTerms={8}
+    handleValueClick={({ generateNextSQON }) => {
+      onValueChange({ sqon: generateNextSQON(sqon) });
+    }}
+    isActive={d =>
+      inCurrentSQON({
+        value: d.value,
+        dotField: d.field,
+        currentSQON: sqon,
+      })
+    }
+    {...{ ...rest }}
```

Comment:
You only need one spread :)

Reply:
lol :P copy-past byproduct :) will clean up